x/lightclient/ante/ibc_msg_update_client.go

```diff
@@ -0,0 +1,96 @@
+package ante
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ibcclienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+	"github.com/dymensionxyz/dymension/v3/x/lightclient/types"
+)
+
+func (i IBCMessagesDecorator) HandleMsgUpdateClient(ctx sdk.Context, msg *ibcclienttypes.MsgUpdateClient) error {
+	clientState, found := i.ibcClientKeeper.GetClientState(ctx, msg.ClientId)
+	if !found {
+		return nil
+	}
+	// Only continue if the client is a tendermint client as rollapp only supports tendermint clients as canonical clients
+	if clientState.ClientType() == exported.Tendermint {
+		// Cast client state to tendermint client state - we need this to get the chain id(rollapp id)
+		tmClientState, ok := clientState.(*ibctm.ClientState)
+		if !ok {
+			return nil
+		}
+		// Check if the client is the canonical client for the rollapp
+		rollappID := tmClientState.ChainId
+		canonicalClient, found := i.lightClientKeeper.GetCanonicalClient(ctx, rollappID)
+		if !found || canonicalClient != msg.ClientId {
+			return nil // The client is not a rollapp's canonical client. Continue with default behaviour.
+		}
+		clientMessage, err := ibcclienttypes.UnpackClientMessage(msg.ClientMessage)
+		if err != nil {
+			return nil
+		}
+		header, ok := clientMessage.(*ibctm.Header)
+		if !ok {
+			return nil
+		}
+		// Check if there are existing block descriptors for the given height of client state
+		height := clientState.GetLatestHeight()
+		stateInfo, err := i.rollappKeeper.FindStateInfoByHeight(ctx, rollappID, height.GetRevisionHeight())
+		if err != nil {
+			// No BDs found for given height.
+			// Will accept the update optimistically
+			// But also save the blockProposer address with the height for future verification
+			blockProposer := header.Header.ProposerAddress
+			i.lightClientKeeper.SetConsensusStateSigner(ctx, msg.ClientId, height.GetRevisionHeight(), string(blockProposer))
+			return nil
+		}
+		bd := stateInfo.GetBDs().BD[height.GetRevisionHeight()-stateInfo.GetStartHeight()]
+
+		ibcState := types.IBCState{
+			Root:               header.Header.GetAppHash(),
+			Height:             height.GetRevisionHeight(),
+			Validator:          header.Header.ProposerAddress,
```

Comment:
Since there will only be one validator, I implemented it this way, but now on rethinking, this has caused me write some extra code to parse the proposer address, and it might be easier to use validatorHash, so will make changes for that 👍🏻

Reply:
implemented validatorsHash checking in https://github.com/dymensionxyz/dymension/pull/1098/commits/61527b81403a9e9d2b36b1e93da3f9836848255f